src/main/java/com/mongodb/kafka/connect/util/ConfigHelper.java

```diff
@@ -48,7 +53,13 @@ private ConfigHelper() {}
             Document.parse(format("{s: %s}", jsonArray)).getList("s", Document.class);
         return s.isEmpty() ? Optional.empty() : Optional.of(s);
       } catch (Exception e) {
-        throw new ConfigException("Not a valid JSON array");
+        if (originalError != null) {
```

Comment:
If the value fails try doubling up on the escaping in the string, if it still fails throw the original error.